### PR TITLE
[Review Replies] Send product review reply to remote

### PR DESCRIPTION
**Pull request by rachelmcr**
Part of #7777
⚠️ Depends on #7789 ⚠️ 

## Description

This PR hooks up the network request to the UI added in #7788 and #7789, to reply to a product review. Now, the reply is sent to the remote site after tapping the "Send" button on the "Reply to Product Review" screen. A loading indicator appears while the network request is being performed, and after it completes a notice appears:

* If the request is successful, the modal is dismissed and a "Reply sent!" notice appears.
* If the request fails, the modal is not dismissed (to avoid losing the reply content) and an error notice appears with a "Retry" action.

## Changes

* Updates `ReviewReplyViewModel` to call the `CommentAction.replyToComment` action when the "Send" button is tapped, and display a notice depending on the result.
* Updates the `ReviewReply` hosting controller to include a modal notice presenter (to display the error notice without dismissing the modal).
* Adds unit tests.

## Testing

Successful reply to a review:

1. Build and run the app in debug mode (to ensure the feature flag is enabled).
2. Tap "Menu" in the bottom navigation bar.
3. Select "Reviews" in the hub menu.
4. Select any product review from the list of reviews.
5. Tap the "Reply" button in the product review detail.
6. Enter some text and tap the "Send" button.
7. Confirm you see the success notice and the reply appears on your store.

Failed reply to review:

1. From the product review detail, tap the "Reply" button again.
2. Disable your network connection to force a network failure.
3. Enter some text and tap the "Send" button.
4. Confirm you see an error notice with a "Retry" button.

## Screenshots

Successful reply (success notice):

https://user-images.githubusercontent.com/8658164/193072691-e36f5c00-a23a-4c09-b0db-e45e5659eefc.mp4

Failed reply (error notice):

https://user-images.githubusercontent.com/8658164/193072713-92dfcaf6-648f-4438-aa92-97fa5a53b9c3.mp4

## Submitter Checklist

Update release notes:
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/Classes/ViewRelated/Reviews/ReviewReplyViewModel.swift [hunk 1]
```diff
@@ -27,8 +52,33 @@ final class ReviewReplyViewModel: ObservableObject {
...
+            self.performingNetworkRequest.send(false)
+
+            switch result {
+            case .success(let status):
+                // If the comment isn't approved, log it (to help support)
+                if status != .approved {
+                    DDLogInfo("Reply to product review succeeded with comment status: \(status)")
+                }
+
+                self.displayReplySuccessNotice()
```
Comment (Ecarrion) [inline WooCommerce/Classes/ViewRelated/Reviews/ReviewReplyViewModel.swift]: Small nit, I think that this probably should be fired from the invoking `ViewController` as it's purely UI logic!

**Review by Ecarrion — APPROVED** [review]
Looks good, just some minor nits!

**Comment by rachelmcr** [comment]
@Ecarrion Thanks for the review! I moved the notice logic into the hosting controller in [ab3ba1c](https://github.com/woocommerce/woocommerce-ios/pull/7791/commits/ab3ba1cc293ee2bebf9addfbdba03425987e3b4d). Since that includes quite a few changes, would you mind taking another look at it?

File: WooCommerce/Classes/ViewRelated/Reviews/ReviewReply.swift [hunk 2]
```diff
@@ -1,21 +1,71 @@
...
 
+        // Observe the present notice intent.
+        presentNoticePublisher
+            .compactMap { $0 }
+            .sink { [weak self] notice in
+                switch notice {
+                case .success:
+                    self?.systemNoticePresenter.enqueue(notice: Notice(title: Localization.success, feedbackType: .success))
+                case .error:
+                    let noticeIdentifier = UUID().uuidString
```
Comment (Ecarrion) [inline WooCommerce/Classes/ViewRelated/Reviews/ReviewReply.swift:53]: Is this to allow multiple error notices to be displayed?

If so: 
1. Neat trick!
2. We should probably add a comment explaining it! 😅

**Review by Ecarrion — APPROVED** [review]

File: WooCommerce/Classes/ViewRelated/Reviews/ReviewReply.swift (hunk 2 above)
Comment (rachelmcr) [reply WooCommerce/Classes/ViewRelated/Reviews/ReviewReply.swift:53]: This is so we can provide `NoticeNotificationInfo` to the `Notice`, so the notice will be presented as a system notification when the app isn't in the foreground:

https://github.com/woocommerce/woocommerce-ios/blob/81ed7b18078c58d3337f53a91d0f09c802bf3c3f/WooCommerce/Classes/Tools/Notices/Notice.swift#L26-L28

I borrowed the idea from our `EditCustomerNote` implementation, with the idea that this way if someone replies to a review and then closes the app, they'll still get a notification if the reply fails.

I'll add a comment to explain!

**Pull request merged**